XMonad/Actions/Menu/KeyParse.hs

```diff
@@ -0,0 +1,347 @@
+module XMonad.Actions.Menu.KeyParse where
+
+{-|
+
+Module      : XMonad.Menu.Actions.KeyParse
+Description : A set of tools to run and render interactive menus
+
+Maintainer  : David Janssen <janssen.dhj@gmail.com>
+Stability   : unstable
+Portability : unportable
+
+This code is taken almost entirely from the key-string parsing capabilities of
+EZConfig, written by Brent Yorgey. EZConfig does not export any of these
+functions and I did not want to modify EZConfig itself, so I extracted the code,
```

Comment:
Hey Brent,

Thanks for the feedback! This is just a quick message to indicate I've seen and read your comments. I'll ponder things over for a bit and reply in detail in the next few days.